src/Nethermind/Nethermind.JsonRpc/Modules/Rbuilder/RbuilderRpcModule.cs

```diff
@@ -0,0 +1,162 @@
+                        {
+                            throw new Exception("Balance is not same!? Why?");
+                        }
+                    }
+                }
+                else
+                {
+                    worldState.CreateAccountIfNotExists(address, accountChange.Balance ?? 0, accountChange.Nonce ?? 0);
+                }
+
+                if (accountChange.CodeHash is not null)
+                {
+                    // Note, this set CodeDb, but since this is a read only world state, it should do nothing.
+                    worldState.InsertCode(address, accountChange.CodeHash, Array.Empty<byte>(), releaseSpec, false);
+                }
+
+                if (accountChange.ChangedSlots is not null)
+                {
+                    foreach (KeyValuePair<UInt256, UInt256> changedSlot in accountChange.ChangedSlots)
+                    {
+                        ReadOnlySpan<byte> bytes = changedSlot.Value.ToBigEndian();
+                        bool newIsZero = bytes.IsZero();
+                        bytes = !newIsZero ? bytes.WithoutLeadingZeros() : [0];
+                        worldState.Set(new StorageCell(address, changedSlot.Key), bytes.ToArray());
+                    }
+                }
+            }
+
+            worldState.Commit(releaseSpec);
+            worldState.CommitTree(blockHeader.Number + 1);
+            return ResultWrapper<Hash256>.Success(worldState.StateRoot);
+        }
+        finally
+        {
+            _overridableWorldScopePool.Return(worldScope);
+        }
+    }
+
+
+    public ResultWrapper<AccountState> rbuilder_getAccount(Address address, BlockParameter block)
+    {
+        BlockHeader? blockHeader = blockFinder.FindHeader(block);
+        if (blockHeader is null)
+        {
+            return ResultWrapper<AccountState>.Fail("Block not found");
+        }
+
+        if (worldStateManager.GlobalStateReader.TryGetAccount(blockHeader.StateRoot!, address,
+                out AccountStruct account))
+        {
+            return ResultWrapper<AccountState>.Success(new AccountState(account.Nonce, account.Balance,
+                account.CodeHash));
+        }
+
+        return ResultWrapper<AccountState>.Success(null);
+    }
+
+    public ResultWrapper<Hash256> rbuilder_getBlockHash(BlockParameter block)
+    {
+
+        BlockHeader? blockHeader = blockFinder.FindHeader(block);
+        return ResultWrapper<Hash256>.Success(blockHeader?.Hash);
+    }
+
+    private class PooledIWorldStatePolicy(IWorldStateManager worldStateManager)
+        : IPooledObjectPolicy<IOverridableWorldScope>
+    {
+        public IOverridableWorldScope Create()
+        {
+            return worldStateManager.CreateOverridableWorldScope();
+        }
+
+        public bool Return(IOverridableWorldScope obj)
+        {
+            obj.WorldState.Reset();
+            obj.WorldState.ResetOverrides();
```

Comment:
Should one invoke the other? Shouldn't there be some unification?

Reply:
@tanishqjasoria is working on a more defined worldstate scope for verkle. I'm thinking of unifying this together with that.